cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -249,6 +249,15 @@ func (t Template) ControllersPolicy() *iamv1.PolicyDocument {
 			})
 		}
 	}
+	if t.Spec.AllowAssumeRole {
+		statement = append(statement, iamv1.StatementEntry{
+			Effect:   iamv1.EffectAllow,
+			Resource: t.allowedEC2InstanceProfiles(),
+			Action: iamv1.Actions{
+				"sts:AssumeRole",
+			},
+		})
+	}
```

Comment:
We have some "assume role" functions, i'm wondering if you could use [AssumeRolePolicy](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/cmd/clusterawsadm/cloudformation/bootstrap/template.go#L234) and pass in `t.allowedEC2InstanceProfiles()`?

Reply:
I dont think that could be done easily as you'd be changing from a principal type to a resource type.  Now I could create another function specifically for passing in a resource instead.

I just took https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go#L217-L223 and stuck it behind a conditional and changed the action.  As mentioned above, if we were to remove the conditional aspect of this, I would have just added the action here and called it a day.  However based on the convo in the Issue linked in earlier comment. with @Skarlso I was under the assumption this needed to be behind a conditional.  So based on that being present as well I could create the new function and have both these statements(Assume/PassRole) be created via the new function instead of how they currently are. 

I really just dont have any skin in the game either way beyond getting the support for the CLI to adding the AssumeRole functionality that we need :)

Reply:
Lets leave it as is :)